File: symfony/src/Events/Tournaments/TournamentEvents.php
<?php

namespace App\Events\Tournaments;

class TournamentEvents {}

final class TournamentCompleted
{
    public function __construct(
        public readonly int $tournamentId,
        public readonly int $seasonId,
        public readonly \DateTimeInterface $completedAt
    ) {}
}

final class PlayerRegistered
{
    public function __construct(
        public readonly int $tournamentId,
        public readonly int $playerId,
        public readonly \DateTimeInterface $registeredAt
    ) {}
}

File: rails/app/events/tournaments/tournament_events.rb
module Events
  module Tournaments
    TournamentCompleted = Struct.new(
      :tournament_id, :season_id, :completed_at,
      keyword_init: true
    )
    PlayerRegistered = Struct.new(
      :tournament_id, :player_id, :registered_at,
      keyword_init: true
    )
  end
end